nilearn/connectivity/connectivity_matrices.py

```diff
@@ -0,0 +1,341 @@
+import warnings
+from math import sqrt
+
+import numpy as np
+from scipy import linalg
+
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.covariance import EmpiricalCovariance
+from .._utils.extmath import is_spd
+
+
+def _check_matrix(matrix, prop):
+    """Raise a ValueError if the input matrix does not satisfy the property.
+
+    Parameters
+    ----------
+    matrix : numpy.ndarray
+        Input array.
+
+    prop : {'square', 'symmetric', 'spd'}
+        Property to check.
+    """
+    if prop == 'square':
+        if matrix.ndim != 2 or (matrix.shape[0] != matrix.shape[-1]):
+            raise ValueError('Expected a square matrix, got array of shape' +
+                             ' {0}.'.format(matrix.shape))
+    if prop == 'symmetric':
+        if not np.allclose(matrix, matrix.T):
+            raise ValueError('Expected a symmetric matrix.')
+
+    if prop == 'spd':
+        if not is_spd(matrix, decimal=7):
+            raise ValueError('Expected a symmetric positive definite matrix.')
+
+
+def _form_symmetric(function, eigen_values, eigen_vectors):
```

Comment:
It seems that eigenvalues and eigenvectors are generally spelled in one word, in which case removing the underscore from these variable names would make sense.